src/components/TabBar.tsx

```diff
@@ -0,0 +1,107 @@
+// 추후 버튼 들어오면 버튼 컴포넌트로 교체 예정
+
+// page에서 렌더링할 컴포넌트를 관리할 state를 선언해주세요 ex) const [ renderedComponent, setRenderedComponent ] = useSTate;
+// page에서 handleTabBarClick 함수를 관리해주세요.
+// const handleTabBarClick = (tabName) => {
+// setRenderComponent(tabName);}
+
+'use client';
+
+import { useState } from 'react';
+
+import classNames from 'classnames';
+import '@/styles/custom.css';
+
+/**
+ * Props for the TabBar component.
+ *
+ * @typedef {Object} TabBarProps
+ * @property {string[]} tabBarList - List of tab names. ex) ['여행 후기', '여행 계획']
+ * @property {function(string): void} handleTabBarClick - Function to handle tab click events. ex) const handleTabBarClick = (tabName) => { setRenderComponent(tabName);}
+ * @property {'S' | 'M' | 'L' | 'vertical_M' | 'vertical_L'} size - Size of the tab bar. 세로 정렬일 경우 vertical 사용
+ */
+
+/**
+ * TabBar component renders a list of tabs.
+ *
+ * @param {TabBarProps} props - The props for the TabBar component.
+ * @returns {JSX.Element} The rendered TabBar component.
+ */
+
+interface TabBarProps {
+  tabBarList: string[];
+  handleTabBarClick: (item: string) => void;
+  size: 'S' | 'M' | 'L' | 'vertical_M' | 'vertical_L';
+}
+
+export default function TabBar({ tabBarList, handleTabBarClick, size }: TabBarProps) {
+  const [clickedTab, setClickedTab] = useState<string>(tabBarList[0]);
+
+  const tabBarCss = {
+    L: {
+      ulClassName: 'tab-bar-ul',
+      liClassName: 'tab-bar-li',
+      hrClassName: 'tab-bar-hr'
+    },
+    M: {
+      ulClassName: 'tab-bar-ul',
+      liClassName: 'tab-bar-li',
+      hrClassName: 'tab-bar-hr'
+    },
+    S: {
+      ulClassName: 'justify-around flex gap-16',
+      liClassName: 'flex flex-col gap-3 w-max',
+      hrClassName: 'tab-bar-hr'
+    },
+    vertical_M: {
+      ulClassName: 'flex flex-col gap-2',
+      liClassName: 'tab-bar-li-vertical-M',
+      hrClassName: 'tab-bar-hr-vertical'
+    },
+    vertical_L: {
+      ulClassName: 'flex flex-col gap-3',
+      liClassName: 'tab-bar-li-vertical-L',
+      hrClassName: 'tab-bar-hr-vertical'
+    }
+  };
+
+  return (
+    <nav>
+      <ul className={tabBarCss[size].ulClassName}>
+        {tabBarList.map((item) => {
+          const className = classNames(
+            {
+              'tab-bar-button-active': clickedTab === item,
+              'tab-bar-button-inactive': clickedTab !== item
+            },
+            'tab-bar-button'
```

Comment:
tab-bar-button-inactive를 기본값으로 주고 active일 때 덮어씌어지게 할 수도 있지 않을까요?

```suggestion
            'tab-bar-button',
            'tab-bar-button-inactive',
            {
              'tab-bar-button-active': clickedTab === item,
            },
```

Reply:
오 넵 가능할 것 같습니다 ! 생각하지 못했던 부분이긴 한데 성능상 큰 차이는 없을 것 같은데, 
가독성 측면이나 유지보수 측면에서 어떤 게 더 나을까요? 
만약 tab-bar-button 클래스 자체를 inactive 클래스랑 합쳐서 두 개만 관리한다면 그게 더 깔끔한 코드일수도 있겠네요!

Reply:
기본값을 정의하고 덮어쓰는 방식이 좀더 가독성부분에서는 좋아보여요! 성능차이는 크게 상관없고 가독성 차이일거같아요ㅎㅎ!

Reply:
inactive가 기본값이라는 생각이 있어서 tab-bar-button 클래스와 합쳐서 관리하는 것도 유지보수 측면에서 괜찮을 거 같습니다.
가독성 측면에서는 tab-bar-button과 tab-bar-button-inactive가 합쳐지지 않는다면 active일 경우, inactive와 active가 같이 있어서 어색하게 느껴질 수도 있을 거 같네요

Reply:
모두 의견 감사합니다! 수정해서 올리겠습니다~